integration-tests/test_client_options.py

```diff
@@ -118,3 +118,27 @@ def test_client_validate_no_network_call(insights_client):
     finally:
         # Remove tags file at the end of test to leave system in clean state
         os.remove(tags_filename)
+
+
+def test_check_show_results(insights_client):
+    """
+    Verify a remediation is returned when checking results from server (--check-results)
+    and then displaying then (--show-results)
+    First I will introduce a vulnerability to the system and at the
+    end of the test I will set the correct file permission again
+    """
+    insights_client.register()
+    assert conftest.loop_until(lambda: insights_client.is_registered)
+
+    try:
+        os.chmod("/etc/ssh/sshd_config", 0o777)
+
+        insights_client.run("--check-results")
+        show_results = insights_client.run("--show-results")
+
+        assert "rule" in show_results.stdout
```

Comment:
Instead of directly looking for text in stdout, I would suggest to look in the json for specific "Problem" related to sshd_config permissions that you introduced. Reason for doing so is to ensure that desired issue is being reported in --show-results(ofcourse there could be multiple advisory but we want to look for the issue  we deliberately introduced):

suggestion:
look for "Update File Permission" , "OPENSSH_HARDENING_CONFIG_PERMS" etc.

Reply:
done